### PR TITLE
KTOR-745 Support null json body

### PR DESCRIPTION
**Pull request by rsinukov**
https://youtrack.jetbrains.com/issue/KTOR-745

### CODE REVIEW

File: ktor-shared/ktor-websocket-serialization/common/src/io/ktor/websocket/serialization/WebsocketChannelSerialization.kt [hunk 1]
```diff
@@ -58,13 +58,18 @@ public suspend inline fun <reified T : Any> WebSocketSession.receiveDeserialized
...
         charset = charset,
-        typeInfo = typeInfo<T>(),
+        typeInfo = typeInfo,
         content = frame
     )
 
     if (result is T) return result
+    if (result == null) {
+        if (typeInfo.kotlinType?.isMarkedNullable == true) return result
+        else throw WebsocketDeserializeException("Frame has null content", frame = frame)
```
Comment (e5l) [inline ktor-shared/ktor-websocket-serialization/common/src/io/ktor/websocket/serialization/WebsocketChannelSerialization.kt:71]: redundant else

File: ktor-client/ktor-client-darwin/darwin/src/io/ktor/client/engine/darwin/DarwinWebsocketSession.kt [hunk 2]
```diff
@@ -188,18 +188,18 @@ internal class DarwinWebsocketSession(
...
-    receiveMessageWithCompletionHandler { message, error ->
-        if (error != null) {
-            it.resumeWithException(DarwinHttpRequestException(error))
-            return@receiveMessageWithCompletionHandler
-        }
-        if (message == null) {
-            it.resumeWithException(IllegalArgumentException("Received null message"))
-            return@receiveMessageWithCompletionHandler
-        }
+private suspend fun NSURLSessionWebSocketTask.receiveMessage(): NSURLSessionWebSocketMessage =
```
Comment (e5l) [inline ktor-client/ktor-client-darwin/darwin/src/io/ktor/client/engine/darwin/DarwinWebsocketSession.kt:191]: could we get to the old formatting? it looks less nested and more readable

**Review by e5l — APPROVED** [review]
lgtm

File: ktor-client/ktor-client-darwin/darwin/src/io/ktor/client/engine/darwin/DarwinWebsocketSession.kt (hunk 2 above)
Comment (rsinukov) [reply ktor-client/ktor-client-darwin/darwin/src/io/ktor/client/engine/darwin/DarwinWebsocketSession.kt:191]: it fails on CI

**Pull request merged**